### PR TITLE
Nc feat/user management

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nocodb/src/helpers/catchError.ts [hunk 1]
```diff
@@ -726,4 +729,10 @@
   static notAllowed(message = 'Not allowed') {
     throw new NotAllowed(message);
   }
+
+  static emailDomainNotAllowed(domain: string) {
+    throw new SsoError(
+      `Email domain ${domain} is not allowed for this organization`,
+    );
+  }
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/helpers/catchError.ts:737]: We are handling this error within global exception handler

File: packages/nc-gui/composables/useOrganization.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  return {
+    orgId,
+    workspaces,
+    listWorkspaces,
+    fetchOrganizationMembers,
+    fetchOrganizationBases,
+    bases,
+    members,
+  }
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nc-gui/composables/useOrganization.ts:22]: It's a cloud-only feature, in OSS we are keeping a placeholder composable.

File: packages/nc-gui/components/workspace/Settings.vue [hunk 3]
```diff
@@ -33,19 +37,29 @@ const formRules = {
...
 
     if (workspacesList.value.length > 1) {
-      await navigateToWorkspace(workspacesList.value[0].id)
+      // WorkspaceId is provided from the admin Panel. If deleted navigate to the workspace list page
+      if (!props.workspaceId) {
+        await navigateToWorkspace(workspacesList.value[0].id)
+      } else {
+        // #TODO: @DarkPhoenix2704
+        // Navigate BackPage
+      }
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nc-gui/components/workspace/Settings.vue:62]: the entire logic is wrapped within a `try...catch` block

File: packages/nc-gui/store/workspace.ts [hunk 4]
```diff
@@ -92,6 +92,8 @@ export const useWorkspace = defineStore('workspaceStore', () => {
 
   const loadWorkspace = async (..._args: any) => {}
 
+  const moveToOrg = async (..._args: any) => {}
+
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nc-gui/store/workspace.ts:96]: It's a placeholder function

File: packages/nc-gui/middleware/02.auth.global.ts [hunk 5]
```diff
@@ -196,3 +198,19 @@ async function tryShortTokenAuth(api: Api<any>, signIn: Actions['signIn']) {
...
+    try {
+      url = decodeURIComponent(window.location.search.split('=')[1])
+    } catch (e: any) {
+      message.error(await extractSdkResponseErrorMsg(e))
+    }
+
+    const newURL = window.location.href.split('?')[0]
+    window.history.pushState('object', document.title, `${newURL}#${url}`)
+    window.location.reload()
+  }
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nc-gui/middleware/02.auth.global.ts:215]: already wrapped with a `try...catch` block

File: packages/nc-gui/components/dlg/InviteDlg.vue [hunk 6]
```diff
@@ -223,40 +262,70 @@ const inviteProjectCollaborator = async () => {
...
-const onRoleChange = (role: keyof typeof RoleLabels) => (inviteData.roles = role as ProjectRoles)
+const organizationStore = useOrganization()
+
+const { listWorkspaces } = organizationStore
+
+const { workspaces } = storeToRefs(organizationStore)
+
+const workSpaceSelectList = computed(() => {
+  return workspaces.value.filter((w) => !workSpaces.value.find((ws) => ws.id === w.id))
+})
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nc-gui/components/dlg/InviteDlg.vue:273]: Will include these improvements in followup PR

**Review by dstala — APPROVED** [review]

**Pull request merged**